File: api/daily-post.py
import json
import os
from http.server import BaseHTTPRequestHandler

from api._bot import TELEGRAM_CHANNEL_ID, generate_post, local_schedule_matches_now, send_message


class handler(BaseHTTPRequestHandler):
    def _send_json(self, status: int, payload: dict) -> None:
        self.send_response(status)
        self.send_header("Content-Type", "application/json")
        self.end_headers()
        self.wfile.write(json.dumps(payload).encode("utf-8"))

    def do_GET(self) -> None:
        cron_secret = os.getenv("CRON_SECRET", "").strip()
        auth = self.headers.get("authorization", "")
        provided = auth.removeprefix("Bearer ").strip()
        if cron_secret and provided != cron_secret:
            self._send_json(401, {"ok": False, "error": "unauthorized"})
            return

        if os.getenv("STRICT_DAILY_SCHEDULE", "0") == "1" and not local_schedule_matches_now():
            self._send_json(200, {"ok": True, "skipped": "outside configured local schedule"})
            return

        if not TELEGRAM_CHANNEL_ID:
            self._send_json(500, {"ok": False, "error": "missing TELEGRAM_CHANNEL_ID"})
            return

        try:
            content, topic = generate_post()
            send_message(TELEGRAM_CHANNEL_ID, content)
            self._send_json(200, {"ok": True, "topic": topic})
        except Exception as exc:
            self._send_json(500, {"ok": False, "error": str(exc)})

File: api/daily-post.js
import { TELEGRAM_CHANNEL_ID, generatePost, saveNotionPost, telegram } from './lib.js';

export default async function handler(req, res) {
  if (req.method !== 'GET' && req.method !== 'POST') {
    return res.status(405).json({ ok: false, error: 'method_not_allowed' });
  }

  if (!TELEGRAM_CHANNEL_ID) {
    return res.status(500).json({ ok: false, error: 'missing_TELEGRAM_CHANNEL_ID' });
  }

  try {
    const { title, topic, content } = await generatePost();
    await telegram('sendMessage', { chat_id: TELEGRAM_CHANNEL_ID, text: content });
    await saveNotionPost({ title, topic, content });
    return res.status(200).json({ ok: true, topic });
  } catch (error) {
    return res.status(500).json({ ok: false, error: error.message });
  }
}